platform/jobs/RetirementJob.groovy

```diff
@@ -0,0 +1,39 @@
+/* UserRetirementJobs.groovy
+ *
+ * Defines jobs which orchestrate user retirement.
+ */
+
+package platform
+
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_TEAM_SECURITY
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_LOG_ROTATOR
+
+// This is the job DSL responsible for creating the main pipeline job.
+pipelineJob('User Retirement') {
+    description('Retire a single user which has confirmed account deletion')
+
+    // Only a subset of edx employees should be allowed to control this job,
+    // but customer support can read and discover.
+    authorization {
+        blocksInheritance(true)
+        List membersWithFullControl = ['edx/platform-team']
+        membersWithFullControl.each { emp ->
+            permissionAll(emp)
+        }
+        // TODO PLAT-2036: uncomment the following two lines when we add the
+        // appropriate github group.
+        //permission('hudson.model.Item.Read', 'edx/customer-support')
+        //permission('hudson.model.Item.Discover', 'edx/customer-support')
```

Comment:
I added this authorization section per your recommendation, but it's quite specific to edX, as you warned.  For now I think it will suffice to get this off the ground.